modules/service/src/main/scala/lucuma/odb/logic/TimeEstimateService.scala

```diff
@@ -163,34 +176,65 @@ object TimeEstimateService {
         root: GroupTree,
         m:    Map[Observation.Id, ObservationData]
       ): F[Option[CategorizedTimeRange]] =
-        root match {
+        root match
           case GroupTree.Leaf(oid)                                  => leafRange(pid, oid, m)
           case GroupTree.Branch(_, min, _, children, _, _, _, _, _) => parentRange(pid, min, children, m)
           case GroupTree.Root(_, children)                          => parentRange(pid, None, children, m)
-        }
 
       private def load(pid: Program.Id): F[(GroupTree, Map[Observation.Id, ObservationData])] =
-        services.transactionally {
+        services.transactionally:
           (groupService.selectGroups(pid), ObservationData.load(pid, itcClient)).tupled
-        }
 
-      override def estimateProgram(
+      override def estimateProgramRange(
         pid: Program.Id
       )(using NoTransaction[F]): F[Option[CategorizedTimeRange]] =
-        load(pid).flatMap { case (tree, dataMap) =>
-          timeEstimateRange(pid, tree, dataMap)
-        }
+        load(pid).flatMap:
+          case (tree, dataMap) => timeEstimateRange(pid, tree, dataMap)
 
-      override def estimateGroup(
-        groupId: Group.Id
+      override def estimateGroupRange(
+        gid: Group.Id
       )(using NoTransaction[F]): F[Option[CategorizedTimeRange]] =
-        (for {
-          p <- OptionT(groupService.selectPid(groupId))
+        (for
+          p <- OptionT(groupService.selectPid(gid))
           d <- OptionT.liftF(load(p))
           (tree, dataMap) = d
-          t <- OptionT.fromOption(tree.findGroup(groupId))
+          t <- OptionT.fromOption(tree.findGroup(gid))
           r <- OptionT(timeEstimateRange(p, t, dataMap))
-        } yield r).value
+        yield r).value
 
-  }
-}
+      def bandedTimeEstimate(
+        pid:  Program.Id,
+        root: GroupTree,
+        m:    Map[Observation.Id, ObservationData]
+      ): F[Map[Option[ScienceBand], CategorizedTime]] =
+
+        val empty: Map[Option[ScienceBand], CategorizedTime] = Map.empty
+
+        def leaf(oid: Observation.Id): F[Map[Option[ScienceBand], CategorizedTime]] =
+          observationTime(pid, oid, m).map: obt =>
+            obt.fold(empty)(bt => Map(bt.band -> bt.time))
+
+        def parent(cs: List[GroupTree.Child]): F[Map[Option[ScienceBand], CategorizedTime]] =
+          cs.traverse(bandedTimeEstimate(pid, _, m)).map(_.foldLeft(empty)(_ |+| _))
```

Comment:
I think `.foldLeft(empty)(a |+| b)` can be written as `.combineAll`

Reply:
Excellent, thanks!